src/components/PrioritySelector/PrioritySelector.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+import Close from '@material-design-icons/svg/round/close.svg';
+import KeyboardDoubleArrowUp from '@material-design-icons/svg/round/keyboard_double_arrow_up.svg';
+import KeyboardArrowUp from '@material-design-icons/svg/round/keyboard_arrow_up.svg';
+import KeyboardArrowDown from '@material-design-icons/svg/round/keyboard_arrow_down.svg';
+import { DropdownMenuItem, Root, Portal } from '@radix-ui/react-dropdown-menu';
+import { DropdownTrigger } from '../Dropdown/DropdownTrigger';
+import { DropdownContent } from '../Dropdown/DropdownContent';
+import { bem } from '../../utils/bem';
+import { Text } from '../Text';
+import { IconButton } from '../Buttons';
+import itemStyles from '../Dropdown/Items/Item.scss';
+import styles from './PrioritySelector.scss';
+
+const iconMap = {
+    mandatory: KeyboardDoubleArrowUp,
+    important: KeyboardArrowUp,
+    optional: KeyboardArrowDown,
+    exclude: Close,
+};
+
+export enum Priority {
+    Mandatory = 'mandatory',
+    Important = 'important',
+    Optional = 'optional',
+    Exclude = 'exclude',
+}
+
+export type PriorityItemType<PriorityItemValue> = {
+    /** priority types: mandatory, important, optional or exclude. Makes correct icon show up */
+    priority: Priority;
+    /** text that should accompany the icon in the dropdown */
+    label: string;
+    /** optional: used in case of application-specific values / custom priorities */
+    value?: PriorityItemValue;
+};
+
+export interface Props<PriorityItemValue>
+    extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onSelect'> {
+    /** Currently selected priority item that indicates the importance of the component. */
+    selectedItem: PriorityItemType<PriorityItemValue>;
+    /** Array of availible priority items. */
+    list: PriorityItemType<PriorityItemValue>[];
+    /** Callback function triggered when a new priority is selected. */
+    onSelect: (newPriorityItem: PriorityItemType<PriorityItemValue>) => void;
+    /** Boolean indicating whether the whole badge should be disabled. */
+    isDisabled?: boolean;
+    /** Priority button label name for ARIA labelling */
+    buttonLabel?: string;
+    /** ref to possible parent div, if it is a badge */
+    parentRef?: React.RefObject<HTMLElement | null> | React.ForwardedRef<HTMLDivElement>;
+    /** ref to iconButton, used for keyboard navigation */
+    buttonRef?: React.RefObject<HTMLElement>;
```

Comment:
```suggestion
    buttonRef?: React.RefObject<HTMLElement>;
    /** the size of the trigger button */
    size?: Size
```
I tried to use in for the Enhanced Button and found that I need an other prop: size. It should render the trigger button smaller or larger (I need smaller) which is already a prop for the Icon button, so just need to pass it through.